paymentsheet/src/main/java/com/stripe/android/paymentsheet/BaseAddPaymentMethodFragment.kt

```diff
@@ -5,284 +5,267 @@ import android.view.LayoutInflater
+                            ?: sheetViewModel.supportedPaymentMethods.first().code
+                    )
+                }
+                val selectedItem = remember(selectedPaymentMethodCode) {
+                    sheetViewModel.supportedPaymentMethods.first {
+                        it.code == selectedPaymentMethodCode
+                    }
+                }
 
-        viewBinding.linkInlineSignup.apply {
-            setContent {
-                val processing by sheetViewModel.processing.observeAsState(false)
+                val layoutFormDescriptor = selectedItem.getPMAddForm(
+                    requireNotNull(sheetViewModel.stripeIntent.value),
+                    sheetViewModel.config
+                )
 
-                PaymentsTheme {
-                    LinkInlineSignup(
-                        linkPaymentLauncher = sheetViewModel.linkLauncher,
-                        enabled = !processing,
-                        onStateChanged = { config, viewState ->
-                            sheetViewModel.updatePrimaryButtonUIState(
-                                if (viewState.useLink) {
-                                    val userInput = viewState.userInput
-                                    if (
-                                        userInput != null &&
-                                        sheetViewModel.selection.value != null
-                                    ) {
-                                        PrimaryButton.UIState(
-                                            label = null,
-                                            onClick = {
-                                                sheetViewModel.payWithLinkInline(
-                                                    config,
-                                                    userInput
-                                                )
-                                            },
-                                            enabled = true,
-                                            visible = true
-                                        )
-                                    } else {
-                                        PrimaryButton.UIState(
-                                            label = null,
-                                            onClick = null,
-                                            enabled = false,
-                                            visible = true
-                                        )
-                                    }
-                                } else {
-                                    null
+                val showLinkInlineSignup = sheetViewModel.isLinkEnabled.value == true &&
+                    sheetViewModel.stripeIntent.value
+                    ?.linkFundingSources?.contains(PaymentMethod.Type.Card.code) == true &&
+                    selectedItem.code == PaymentMethod.Type.Card.code &&
+                    linkAccountStatus == AccountStatus.SignedOut
```

Comment:
This is pretty busy. As a follow-up, we should also move over to a view state for the `PaymentElement`.